### PR TITLE
AFS add_weights* functions updates

### PR DESCRIPTION
**Pull request by aspina7**
Just adding some errors/warnings to make these more user friendly. 

I guess some of the errors could be put in a loop but not sure how?

### CODE REVIEW

File: R/add_weights_cluster.R [hunk 1]
```diff
@@ -119,19 +119,86 @@ add_weights_cluster <- function(x, cl,
...
+  indiv_surv <- tidyselect::vars_select(colnames(x),  {{interviewed}}, .strict = FALSE)
+
+  ## checks (errors / warnings)
+
+  # check if variables are in datasets (spelling) - throws error
+
+  # for cluster dataset
+  if (length(clus_id_cl) == 0) {
+    # pull the environment from which this was called (i.e. the function)
+    cll   <- match.call()
```
Comment (zkamvar) [inline R/add_weights_cluster.R:142]: I would suggest to move the `match.call()` to the top of the function. You can refer to it in each of these errors later. That being said, it's not a deal breaker.

File: R/add_weights_cluster.R [hunk 2]
```diff
@@ -119,19 +119,86 @@ add_weights_cluster <- function(x, cl,
...
+  }
+
+
+  # check if there are duplicate cluster names (i.e. counted twice) - throws error
+  dup_clus <- cl[duplicated(cl[ , clus_id_cl]) , clus_id_cl]
+
+  if (length(dup_clus) > 0) {
+    cll   <- match.call()
+    stop(glue::glue("Cluster names are duplicated in {clus_id_cl} variable of {cll[['cl']]} dataset"))
+  }
```
Comment (zkamvar) [inline R/add_weights_cluster.R:188]: This is clearer and avoids using matrix subsetting

```suggestion
  if (any(duplicated(cl[[clus_id_cl]])) {
    cll   <- match.call()
    stop(glue::glue("Cluster names are duplicated in {clus_id_cl} variable of {cll[['cl']]} dataset"))
  }
```

File: R/add_weights_cluster.R [hunk 3]
```diff
@@ -119,19 +119,86 @@ add_weights_cluster <- function(x, cl,
...
+
+  # check how many cluster names in the study data match cluster counts data
+  matching_clusters <- unique(x[[clus_id_x]][which(!x[[clus_id_x]] %in% cl[[clus_id_cl]])])
+
+  if (length(matching_clusters) > 0) {
+    cll <- match.call()
+    warning(glue::glue("The following cluster names in the {clus_id_x} variable are not in the {cll[['cl']]} dataset: ",
+                       glue::glue_collapse(matching_clusters, sep = ",")))
+  }
```
Comment (zkamvar) [inline R/add_weights_cluster.R:197]: Breaking this up and renaming for clarity
```suggestion
  non_matching_clusters <- !x[[clus_id_x]] %in% cl[[clus_id_cl]]

  if (any(non_matching_clusters)) {
    nm_cluster_names <- unique(x[[clus_id_x]][non_matching_clusters])
    cll <- match.call()
    warning(glue::glue("The following cluster names in the {clus_id_x} variable are not in the {cll[['cl']]} dataset: ",
                       glue::glue_collapse(nm_cluster_names, sep = ",")))
  }
```

**Review by zkamvar — COMMENTED** [review]
Looks good to me. We could stand to add tests for these warnings/errors, but it's fine as is.

File: R/add_weights_cluster.R (hunk 1 above)
Comment (aspina7) [reply R/add_weights_cluster.R:142]: yes please i just couldnt check if it worked when i was writing it - but if it works then cool

Comment (aspina7) [reply R/add_weights_cluster.R]: fixed in latest commit

**Comment by aspina7** [comment]
@zkamvar this one is also independent of others ... have address the match.call() thingy, so hopefully things should be working. Theres another issue open on shifting these functions to {epikit} but we will get to that. 
For now can we merge please - so can close relevant issues? thx!

**Comment by aspina7** [comment]
> @zkamvar this one is also independent of others ... have address the match.call() thingy, so hopefully things should be working. Theres another issue open on shifting these functions to {epikit} but we will get to that. For now can we merge please - so can close relevant issues? thx!

@zkamvar just poking again incase you have a min inbetween

**Comment by aspina7** [comment]
@nsbatra and @finlaycampbell - if you have some time could you review/trial these functions please? 

They are for adding weights to individuals in survey data - to be able to correct for sampling based on population counts. 
Weights are used to feed in to {survey} (or {srvyr}) objects for producing weighted counts and proportions (i.e. they are essentially a multiplier). 

There are two functions - one for [stratified survey designs ](R/add_weights_strata.R) and one for [cluster survey designs](R/add_weights_cluster.R) (weights for stratified clusters just involves multiplying the two output values). 
There are [tests for cluster weighting](tests/testthat/test-add_weights_cluster.R); and just noticed that probably need to add for strata too... 

See this [code chunk](https://github.com/R4EPI/sitrep/blob/master/inst/rmarkdown/templates/mortality/skeleton/skeleton.Rmd#L749) in mortality survey template for how they are used in practice and the following chunk for how weights are fed to {srvyr} functions. 

After merge Im going to move these functions to {epikit} as per #269 

Thanks!

**Comment by aspina7** [comment]
going to merge for now to get started on with #134 and #269

**Pull request merged**